cryptographic_flows.md

```diff
@@ -0,0 +1,27 @@
+
+# Authenticate
+The local client inputs a user ID and a password and receives an "export key", which we will consider as a valid symmetric encryption key, as output. The key server receives as output some user-specific authentication material, which it can compare to its stored data to verify authentication.
+
+Both parties recieve an open channel as output, which satisfies mutual authentication, confidentiality, and integrity.
+
+
+# Generate a secret
+
+The local client component can generate a secret. Using a cryptographically secure pseudo-random number generator with a random seed, generate 256 bits of output.
+
+Generate a unique tag to associate with the secret (unique among secrets stored by the local client).
+
+# Generate and store a secret locally
+The local client component can run the [generate](generate-a-secret) protocol. Then, store the tag and the secret as a pair.
+
+# Generate and store a secret remotely
+The local client must
+1. Run the [authenticate](authenticate) protocol to retrieve an export key and an open channel.
```

Comment:
- Reference the relevant issue (#4) here for authenticate, rather than including an imprecise description above.
- I really think we want to handle session abstractions separately, e.g., this isn't really what we want to happen in the implementation, is it? What if the user already has an open authenticated session and they go to generate and store a secret at the key server? It seems that we want to specify expected behavior that captures this scenario as well.

Reply:
I changed this to just say that the session must already be open, but without the specification that it has to be step 1 here.

Reply:
That sounds too vague, then. It has to be the first thing.